go/enclave/nodetype/sequencer.go

```diff
@@ -263,26 +285,26 @@ func (s *sequencer) duplicateBatches(l1Head *types.Block, nonCanonicalL1Path []c
 		return batchesToDuplicate[i].Number().Cmp(batchesToDuplicate[j].Number()) == -1
 	})
 
-	sequencerNo, err := s.storage.FetchCurrentSequencerNo()
-	if err != nil {
-		return fmt.Errorf("could not fetch sequencer no. Cause %w", err)
-	}
-
 	currentHead := batchesToDuplicate[0].Header.ParentHash
 
 	// find all batches for that path
 	for i, orphanBatch := range batchesToDuplicate {
 		// sanity check that all these batches are consecutive
-		if i > 0 && !bytes.Equal(batchesToDuplicate[i].Header.ParentHash.Bytes(), batchesToDuplicate[i-1].Hash().Bytes()) {
+		if i > 0 && batchesToDuplicate[i].Header.ParentHash != batchesToDuplicate[i-1].Hash() {
 			s.logger.Crit("the batches that must be duplicated are invalid")
 		}
+		sequencerNo, err := s.storage.FetchCurrentSequencerNo()
+		if err != nil {
+			return fmt.Errorf("could not fetch sequencer no. Cause %w", err)
+		}
 		sequencerNo = sequencerNo.Add(sequencerNo, big.NewInt(1))
+		seqNoCp := *sequencerNo
```

Comment:
This pointer operation doesn't help for big int - the struct contains pointers so a shallow copy isn't safe.

Not sure if this is still needed after you found the other problem, but if it is you can use

`seqNoCp := new(big.Int).Set(sequencerNo)`

to do a deep copy (a source: https://groups.google.com/g/golang-nuts/c/skZ5J7gEVcU)

Reply:
that explains why I had to move the query inside the for loop.